azimuth/plots/dataset_warnings.py

```diff
@@ -437,6 +406,78 @@ def create_histogram_mean_std(
             yanchor="middle",
             showarrow=False,
         )
+    else:
+        y_per_split = {DatasetSplitName.eval: 1.14, DatasetSplitName.train: 1.07}
+        opacity_per_split = {DatasetSplitName.eval: 1, DatasetSplitName.train: 0.5}
+        # sorted because the color overlay looks better when eval is first
+        for split, hist in sorted(hist_normalized.items()):
+            fig.add_bar(
+                x=list(range(1, len(hist) + 1)),
+                y=hist,
+                name=DATASET_SPLIT_PRETTY_NAMES[split],
+                marker=dict(color=DATASET_SPLIT_COLORS[split], opacity=opacity_per_split[split]),
+            )
+
+            fig.add_scatter(
+                x=[np.round(value_per_agg_per_split[split][Agg.mean], 2)],
+                y=[max_y * y_per_split[split]],
+                name=f"{split}_mean_std",
+                error_x=dict(
+                    type="constant", value=np.round(value_per_agg_per_split[split][Agg.std], 2)
+                ),
+                hoverinfo="x",
+                marker=dict(color=DATASET_SPLIT_COLORS[split]),
+            )
+
+        if divergence_per_agg and len(hist_normalized) == 2:
+            fig.add_annotation(
+                x=0,
+                y=max_y * 1.2,
+                text=f"Delta of {divergence_per_agg[Agg.mean]:.2f}±"
+                f"{divergence_per_agg[Agg.std]:.2f} tokens",
+                font=dict(color=Colors.Text),
+                xref="paper",
+                xanchor="left",
+                showarrow=False,
+            )
+
+        # Add thresholds for short and long utterances.
+        max_x = max((len(hist) for hist in hist_normalized.values()), default=0)
+        for x0, x1, pretty_name, xanchor in zip(
+            [0, float(syntax_options.long_utterance_min_word - 0.5)],
+            [float(syntax_options.short_utterance_max_word) + 0.5, max_x + 1],
+            ["Short utterances", "Long utterances"],
+            ["right", "left"],
+        ):
+            threshold = x1 if xanchor == "right" else x0
+            common_args = dict(layer="below", y0=0, y1=max_y)
+            fig.add_shape(
+                type="rect",
+                x0=x0,
+                x1=x1,
```

Comment:
I wouldn't be surprised if `x0` and `x1` were completely interchangeable (like to constraint on `x0` < `x1`), so you could zip directly on `threshold` and say `edge`
```diff
-        for x0, x1, pretty_name, xanchor in zip(
-            [0, float(syntax_options.long_utterance_min_word - 0.5)],
-            [float(syntax_options.short_utterance_max_word) + 0.5, max_x + 1],
+        for threshold, edge, pretty_name, xanchor in zip(
+            [syntax_options.short_utterance_max_word + 0.5, syntax_options.long_utterance_min_word - 0.5],
+            [0, max_x + 1],
```
to avoid
```diff
-            threshold = x1 if xanchor == "right" else x0
```
and do
```suggestion
                x0=threshold,
                x1=edge,
```
but I haven't tested it. No big deal, of course, just if you feel like it.

Reply:
I don't think this would work for the added vertical line and annotation right?

Reply:
I think it would, `threshold` is, well, the threshold, and `edge` is the outer edge of the rectangle. (you could rename it if you find something better). Am I missing something?

Reply:
This for loop creates the 2 rectangles. The threshold is the right side for the left rectangle, and the left side for the right rectangle.

Reply:
Correct, so what I'm saying is that I think we are not restricted to `x0` < `x1`, so `x0` can always be `threshold` and `x1` the other edge, no matter if it's to the left or to the right.

Reply:
Right, I think my brain is finally joining the thread. It should work, I'll test it. :)

Reply:
Haha! 😝 Cool!